lib/iris/tests/unit/coords/test_Coord.py

```diff
@@ -355,6 +356,112 @@ def test_different_array_attrs_incompatible(self):
         self.assertFalse(self.test_coord.is_compatible(self.other_coord))
 
 
+class Test_contiguous_bounds(tests.IrisTest):
+    def test_no_bounds_warning(self):
+        coord = iris.coords.DimCoord([0, 1, 2], standard_name='latitude')
+        msg = "Coordinate 'latitude' is not bounded, guessing contiguous " \
+              "bounds."
+        with warnings.catch_warnings():
+            # Cause all warnings to raise Exceptions
+            warnings.simplefilter("error")
+            with self.assertRaisesRegexp(Warning, msg):
+                coord.contiguous_bounds()
+
+    def test_1d_coord(self):
+        coord = iris.coords.DimCoord([2, 4, 6], standard_name='latitude',
+                                     bounds=[[1, 3], [3, 5], [5, 7]])
+        expected = np.array([1, 3, 5, 7])
+        result = coord.contiguous_bounds()
+        self.assertArrayEqual(result, expected)
+
+    def test_1d_coord_discontiguous(self):
+        coord = iris.coords.DimCoord([2, 4, 6], standard_name='latitude',
+                                     bounds=[[1, 3], [4, 5], [5, 7]])
+        msg = "Invalid operation. Bounds of coord 'latitude' are not " \
+              "contiguous."
+        with self.assertRaisesRegex(ValueError, msg):
+            coord.contiguous_bounds()
+
+
+class Test_is_contiguous(tests.IrisTest):
+    def setUp(self):
+        self.points = np.array([1, 3])
+        self.bounds = np.array([[0, 2], [2, 4]])
+
+    def test_no_bounds(self):
+        coord = iris.coords.DimCoord(self.points)
+        result = coord.is_contiguous()
+        self.assertFalse(result)
+
+    def test_1d_coord_contiguous(self):
+        coord = iris.coords.DimCoord(self.points, bounds=self.bounds)
+        result = coord.is_contiguous()
+        self.assertTrue(result)
+
+    def test_1d_coord_discontiguous(self):
+        coord = iris.coords.DimCoord(self.points, bounds=[[0, 2], [3, 4]])
+        result = coord.is_contiguous()
+        self.assertFalse(result)
+
+    def test_1d_coord_contiguous_atol(self):
+        coord = iris.coords.DimCoord(self.points/1000, bounds=self.bounds/1000)
+        result = coord.is_contiguous(atol=1e-2)
+        self.assertTrue(result)
+
+
+class Test__sanity_check_bounds(tests.IrisTest):
+    def test_coord_1d_2_bounds(self):
+        coord = iris.coords.DimCoord([0, 1], standard_name='latitude',
+                                     bounds=np.array([[0, 1], [1, 2]]))
+        coord._sanity_check_bounds()
```

Comment:
This is just to check that no error is raised is it?

Reply:
Yep I'll add a comment explaining that